### ISSUE TITLE
False positive for unused variable when using new format style

### ISSUE THREAD

**Issue by octylFractal** [intellij-rust/intellij-rust#8417]
## Environment

* **IntelliJ Rust plugin version:** 0.4.164.4374-213-nightly
* **Rust toolchain version:** 1.58.0 (02072b482 2022-01-11) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.3.2 (CL-213.6461.75)
* **Operating system:** Linux 5.15.12-zen1-1-zen
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
The `args` variable below is marked as unused, even though it's certainly used by the `println!`.

I've tried invalidating caches, but it didn't help.

## Steps to reproduce
Just make this the content of a file in CLion, and it will immediately be marked as unused.
```rust
fn main() {
    let args = 4;
    println!("{args}");
}
```

**Comment by Undin** [comment]
Part of #7338

**Comment by grainednoise** [comment]
For what it's worth, I run into this issue a lot, to the point that it has almost become my pet annoyance.

**Comment by mellonges** [comment]
it would be nice to add an autocomplete inside the template {}

**Comment by KisaragiEffective** [comment]
> it would be nice to add an autocomplete inside the template {}

you should create new issue

**Comment by Miha-Rozina** [comment]
Is this issue being worked on? It makes files without warnings seem like they have warnings. And it makes the yellow lines in the scroll bar useless since there is so many false positives in some files.

**Comment by Kobzol** [comment]
https://github.com/intellij-rust/intellij-rust/pull/8561 I tried to draft something, but it's not trivial to change this in the plugin, so it might take some time.
